src/Module.php

```diff
@@ -104,10 +108,17 @@ public function add_zone_backend(string $domain, string $ip): bool
 			 */
 			try
 			{
+				$kind = 'native'; // Enables backend replication via MySQL Replication or MariaDB Galera replication
+
+				if (defined('AUTH_PDNS_TYPE') && in_array(AUTH_PDNS_TYPE, static::$permitted_types))
```

Comment:
Use strict type checking, i.e. `in_array(AUTH_PDNS_TYPE, static::$permitted_types, true)`. Use strict typing unless you're knowingly juggling multiple types as with parameter types (int|bool).

Also, it's a little more clear if the type is invalid

```php
if (defined('AUTH_PDNS_TYPE')) {
    if (!in_array(AUTH_PDNS_TYPE, static::$permitted_types, true)) {
        fatal("Unknown PowerDNS server type `%s'", AUTH_PDNS_TYPE);
    }
    $kind = AUTH_PDNS_TYPE;
}